contracts/modules/common/SimpleOracle.sol

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.s
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router01.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Pair.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+contract SimpleOracle {
+
+    using SafeMath for uint256;
+
+    address immutable weth;
+    address immutable factory;
+
+    constructor(address _uniswapRouter) public {
+        weth = IUniswapV2Router01(_uniswapRouter).WETH();
+        factory = IUniswapV2Router01(_uniswapRouter).factory();
+    }
+
+    function inToken(address _token, uint256 _ethAmount) internal view returns (uint256) {
+        (uint256 wethReserve, uint256 tokenReserve) = getReservesForTokenPool(_token);
+        return _ethAmount.mul(tokenReserve) / wethReserve;
+    }
+
+    function getReservesForTokenPool(address _token) internal view returns (uint256 wethReserve, uint256 tokenReserve) {
+        if (weth < _token) {
+            address pair = getPairForSorted(weth, _token);
+            (wethReserve, tokenReserve,) = IUniswapV2Pair(pair).getReserves();
+        } else {
+            address pair = getPairForSorted(_token, weth);
+            (tokenReserve, wethReserve,) = IUniswapV2Pair(pair).getReserves();
+        }
+        require(wethReserve != 0 && tokenReserve != 0, "SO: no liquidity");
```

Comment:
Unlikely scenario but if the liquidity is removed, perhaps we should skip the refund (i.e. return 0 here) instead of reverting the whole transaction.

Reply:
I would prefer to be strict so it cannot be abused. If there is no liquidity then we should not let the user refund with that token (i.e. the transaction should not be relayed) but if it happens then the transaction should fail.

However, you raise a valid point: what happens when Uniswap releases its V3?